lib/io.js

```diff
@@ -61,14 +56,7 @@ BinaryDecoder.prototype = {
         var numShifts = 0;
 
         while (b.greaterThan(L_0x7f) && numShifts < 7) {
-            b = this.readByte();
-            if (b instanceof AvroErrors.BlockDelayReadError) {
-                //console.error("offset now is %d, old was %d, shift is %d", this._input.offset, oldOffset, shift);
-                this._input.skip(oldOffset - this._input.offset);
```

Comment:
Yes, I modified datafile.js to throw the error rather than returning it (this is where the error originates).